lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -855,6 +855,140 @@ static void optimizeBatchNorm(Function *F) {
   } // For all nodes in the graph.
 }
 
+/// Check if the \p inputs of this method (vector of NodeValues) can be
+/// legal input of CreateConcat function.
+/// \returns the dimension where the Concat will take place. If not
+/// concatenatable, return -1.
+static int
+getConcatDimensionForInputs(const llvm::SmallVectorImpl<NodeValue> &inputs,
```

Comment:
This function is way too long. Can you break it up to utilities that check some properties of shapes?